### PR TITLE
DOC, BLD: fix doc build issues in preparation for the next numpydoc release

### PR DESCRIPTION
**Pull request by rgommers**
The main issue was that numpydoc got really picky about `See Also` sections. Also I managed to track down the last warnings that still show up in the CI doc build.

The ones that don't show up in the CI build but I can't get rid of no matter my version of numpydoc or removing `__call__` from `doc/source/_templates/autosummary/class.rst` are these:
```
/home/rgommers/code/numpy/doc/source/reference/routines.polynomials.polynomial.rst:23:<autosummary>:: WARNING: toctree contains reference to nonexisting document 'reference/generated/numpy.polynomial.polynomial.Polynomial.__call__'
/home/rgommers/code/numpy/doc/source/reference/routines.testing.rst:56:<autosummary>:: WARNING: toctree contains reference to nonexisting document 'reference/generated/numpy.testing.suppress_warnings.__call__'
```

At least there's no difference between numpydoc versions, and the generated docs look fine.

EDIT: for the See Also changes of generic, here's an example of how they are rendered incorrectly at the moment: https://www.numpy.org/devdocs/reference/generated/numpy.generic.clip.html#numpy.generic.clip

### CODE REVIEW

File: numpy/lib/function_base.py [hunk 1]
```diff
@@ -2602,7 +2602,7 @@ def blackman(M):
     Plot the window and the frequency response:
 
     >>> from numpy.fft import fft, fftshift
-    >>> window = np.blackman(51)
+    >>> window = np.blackman(51) + 1e-10  # eps shift to avoid division by zero
```
Comment (eric-wieser) [inline numpy/lib/function_base.py]: What happened without this? Which line causes the divide by 0?

**Comment by mattip** [comment]
@rgommers I don't see warnings with cpython 3.6.7, docutils 0.14, sphinx 1.8.5 and running `git clean -xfd`

File: numpy/lib/function_base.py (hunk 1 above)
Comment (rgommers) [reply numpy/lib/function_base.py]: this generates a window with the first element 0, and that's fed into `log10` which generates a warning

Comment (eric-wieser) [reply numpy/lib/function_base.py]: "division by zero" is a weird way to describe `log10(0)`. How about `logarithm of zero`?

Comment (eric-wieser) [reply numpy/lib/function_base.py]: It's worse than that, the first element is actually negative

Comment (eric-wieser) [reply numpy/lib/function_base.py]: Given that clipping occurs on line 2622, I think it would be better to use the `errstate` context manager to silence the warning on the line that performs the `log`

**Comment by mattip** [comment]
It seems this PR doesn't actually update the numpydoc repo to latest master. When I do that then try to build docs, I get errors that are hard to decipher due to "See also the corresponding ..." in the docstring for `generic.T`. This seems like a bug in numpydoc. The regex should be backward compatible, and should produce a more helpful error message.

**Comment by rgommers** [comment]
@mattip no it doesn't. I'm in the middle of cutting a new numpydoc release and am testing with numpydoc master plus some yet to be merged branches. So I just want to fix things that need fixing here; numpydoc will still change over the coming days so don't want to update its version in this PR.

**Comment by rgommers** [comment]
> I get errors that are hard to decipher due to "See also the corresponding ...

You are testing not with this branch. The fix for what you are reporting is already in this PR. And it does need fixing in numpy because what we currently have looks bad: https://www.numpy.org/devdocs/reference/generated/numpy.generic.clip.html#numpy.generic.clip

**Comment by mattip** [comment]
I would claim the current rendering is a bug with numpydoc, and not with the documentation. Documentation should be able to use the phrase "See also the" without it being flagged as a directive

**Comment by rgommers** [comment]
No, See Also is and has also always been for linking, not for general descriptions. The spec is also pretty clear on this point: https://numpydoc.readthedocs.io/en/latest/format.html#sections

**Comment by mattip** [comment]
Sorry, my bad. I had a mix of this branch with an older version of numpy. I wonder if there is a way to get the makefile to check the imported numpy version vs. the expected one, which might help some of the time. It would not pick up local changes but would catch gross errors like mine

**Comment by rgommers** [comment]
> I wonder if there is a way to get the makefile to check the imported numpy version vs. the expected one

well what is the expected one? the `make dist` command ensures you get the right version because it does a local rebuild of the current state of the git repo first. we could add something like checking if `../numpy/version.py` exists, but also that is not guaranteed to contain the correct git hash (and it would be a bit hacky...).

File: numpy/lib/function_base.py (hunk 1 above)
Comment (rgommers) [reply numpy/lib/function_base.py]: Good suggestion, that is cleaner. updated

File: numpy/ma/extras.py [hunk 2]
```diff
@@ -1532,7 +1532,7 @@ def flatnotmasked_edges(a):
 
     See Also
     --------
-    flatnotmasked_contiguous, notmasked_contiguous, notmasked_edges,
+    flatnotmasked_contiguous, notmasked_contiguous, notmasked_edges
```
Comment (mattip) [inline numpy/ma/extras.py]: The [rendered document](https://6741-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.ma.flatnotmasked_edges.html?highlight=notmasked_edges#numpy.ma.flatnotmasked_edges) adds the comma back? Nice.

File: numpy/lib/function_base.py [hunk 3]
```diff
@@ -1929,6 +1929,30 @@ class vectorize(object):
...
+    calls, so only do this if your function is expensive.
+
+    The new keyword argument interface and `excluded` argument support
+    further degrades performance.
+
+    References
+    ----------
+    .. [1] NumPy Reference, section `Generalized Universal Function API
+           <https://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html>`_.
+
```
Comment (mattip) [inline numpy/lib/function_base.py]: Not part of the PR, but maybe this reference should be a `See Also`. It is [rendered](https://6741-908607-gh.circle-artifacts.com/0/home/circleci/repo/doc/build/html/reference/generated/numpy.vectorize.html) with some strange text in place of `[1]`

**Comment by mattip** [comment]
Thanks @rgommers
